library/ssl_tls.c

```diff
@@ -5120,15 +5122,23 @@ static int ssl_compute_master( mbedtls_ssl_handshake_params *handshake,
         size_t other_secret_len = 0;
         unsigned char* other_secret = NULL;
 
-        if ( handshake->ciphersuite_info->key_exchange == MBEDTLS_KEY_EXCHANGE_RSA_PSK )
+        switch( handshake->ciphersuite_info->key_exchange )
         {
-            /* Provide other key as other secret.
-             * For RSA-PKS other key length is always 48 bytes.
+            /* Provide other secret.
              * Other secret is stored in premaster, where first 2 bytes hold the
-             * length of the other key. Skip them.
+             * length of the other key.
              */
-            other_secret_len = 48;
-            other_secret = handshake->premaster + 2;
+            case MBEDTLS_KEY_EXCHANGE_RSA_PSK:
+                /* For RSA-PKS other key length is always 48 bytes. */
+                other_secret_len = 48;
+                other_secret = handshake->premaster + 2;
+                break;
+            case MBEDTLS_KEY_EXCHANGE_ECDHE_PSK:
+                other_secret_len = MBEDTLS_GET_UINT16_BE(handshake->premaster, 0);
```

Comment:
Actually we could merge all cases, as for RSA-PSK, the value 48 will have been written to the first two bytes.

Reply:
Nope, sorry, it would have been written by `mbedtls_ssl_psk_derive_premaster()` but we're skipping it for opaque. I'm glad I'm making that second pass.

Reply:
Yes, I also tried to remove hard coded 48 bytes for RSA-PSK, but it turned out that it is needed (at least for now).